scipy/integrate/_quadrature.py

```diff
@@ -822,6 +825,270 @@ def simpson(y, *, x=None, dx=1.0, axis=-1, even=_NoValue):
+    initial : scalar or array_like, optional
+        If given, insert this value at the beginning of the returned result,
+        and add it to the rest of the result. Default is None, which means no
+        value at ``x[0]`` is returned and `res` has one element less than `y`
+        along the axis of integration. Can either be a float, or an array with
+        the same shape as `y`, but length 1 along `axis`.
+
+    Returns
+    -------
+    res : ndarray
+        The result of cumulative integration of `y` along `axis`.
+        If `initial` is None, the shape is such that the axis of integration
+        has one less value than `y`. If `initial` is given, the shape is equal
+        to that of `y`.
+
+    See Also
+    --------
+    numpy.cumsum
+    cumulative_trapezoid : cumulative integration using the composite 
+        trapezoidal rule
+    simpson : integrator for sampled data using the Composite Simpson's Rule
+
+    Notes
+    -----
+
+    .. versionadded:: 1.12.0
+
+    The composite Simpson's 1/3 method can be used to approximate the definite 
+    integral of a sampled input function :math:`y(x)` [1]_. The method assumes 
+    a quadratic relationship over the interval containing any 3 consecutive 
+    sampled points. The integral is estimated separately for each interval
+    and then cumulatively summed to obtain the final result.
+
+    Consider three consecutive points: 
+    :math:`(x_1, y_1), (x_2, y_2), (x_3, y_3)`.
+    where the widths of the 2 subintervals are:
+    :math:`h_1 = x_2 - x_1` and :math:`h_2 = x_3 - x_2`.
+
+    Assuming a quadratic relationship over the 3 points, the following 
+    integrals for the subintervals are derived (formulae (6) and (8) of [2]_):
+
+    .. math::
+        \int_{x_1}^{x_3} y(x) dx\ = \frac{x_3-x_1}{6}\left[\
+        \left\{2-\frac{x_3-x_2}{x_2-x_1}\right\} y_1 + \
+        \frac{(x_3-x_1)^2}{(x_2-x_1)(x_3-x_2)} y_2 + \
+        \left\{2-\frac{x_2-x_1}{x_3-x_2}\right\} y_3\right]
+
+    .. math::
+        \int_{x_1}^{x_2} y(x) dx\ = \frac{x_2-x_1}{6}\left[\
+        \left\{3-\frac{x_2-x_1}{x_3-x_1}\right\} y_1 + \
+        \left\{3 + \frac{(x_2-x_1)^2}{(x_3-x_2)(x_3-x_1)} + \
+        \frac{x_2-x_1}{x_3-x_1}\right\} y_2 - \
+        \frac{(x_2-x_1)^2}{(x_3-x_2)(x_3-x_1)} y_3\right]
+
+    .. math::
+        \int_{x_2}^{x_3} y(x) dx\ = \frac{x_3-x_2}{6}\left[\
+        - \frac{(x_3-x_2)^2}{(x_2-x_1)(x_3-x_1)} y_1 + \
+        \left\{3 + \frac{(x_3-x_2)^2}{(x_2-x_1)(x_3-x_1)} + \
+        \frac{x_3-x_2}{x_3-x_1}\right\} y_2 + \
+        \left\{3-\frac{x_3-x_2}{x_3-x_1}\right\} y_3\right]
+
+    Note that the integral formulae in [1]_ and [2]_ use different variables.
+
+    For samples that are equally spaced, the result is exact if the function
+    is a polynomial of order 3 or less [1]_. If the samples are not equally 
+    spaced, then the result is exact only if the function is a polynomial of 
+    order 2 or less.
```

Comment:
Doesn't this more closely match what we are actually doing, though?

```suggestion
    Assuming a quadratic relationship over the 3 points, the integral over 
    the subinterval between :math:`x_1` and :math:`x_2` is given by formula
    (8) of [2]_:
    
    .. math::
        \int_{x_1}^{x_2} y(x) dx\ = \frac{x_2-x_1}{6}\left[\
        \left\{3-\frac{x_2-x_1}{x_3-x_1}\right\} y_1 + \
        \left\{3 + \frac{(x_2-x_1)^2}{(x_3-x_2)(x_3-x_1)} + \
        \frac{x_2-x_1}{x_3-x_1}\right\} y_2 - \
        \frac{(x_2-x_1)^2}{(x_3-x_2)(x_3-x_1)} y_3\right]

    The integral between :math:`x_2` and :math:`x_3` is given by swapping
    appearances of :math:`x_1` and :math:`x_3`.
    
    For samples that are equally spaced, the result is exact if the function
    is a polynomial of order 3 or less [1]_ and the number of subintervals
    is even. Otherwise, the integral is exact for polynomials of  order 2 or
    less.
```

I don't think we need to mention the integral between `x1` and `x3` because you already said that the overall integral is calculated by cumulative summation over the sub-intervals.